<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="publish_tf" default="true"/>
    <arg name="sensor_rate" default="10.0"/>
    <arg name="use_world_frame" default="true"/>
	<arg name="pointcloud_topic" default="/ouster_points_self_filtered"/>

    <group if="$(arg publish_tf)">
        <include file="$(find m545_urdf)/launch/load.launch"/>
        <include file="$(find m545_rviz)/launch/tf_publisher.launch"/>
    </group>
    
    <include file="$(find ground_remover)/launch/ground_remove.launch">
        <arg name="input_topic" value="/dynamic_mapping/raw_pointcloud"/>
        <arg name="algorithm" value="0"/>
    </include>
    <node name="dynamic_mapping" pkg="dynamic_mapping" type="dynamic_mapping_node.py" output="screen">
        <param name="sensor_frequency" type="double" value="$(arg sensor_rate)"/>
		<param name="use_world_frame" type="bool" value="$(arg use_world_frame)"/>
		<param name="pointcloud_topic" type="string" value="$(arg pointcloud_topic)"/>
    </node>
</launch>
